nexus_workcell_orchestrator/src/workcell_orchestrator.cpp

```diff
@@ -895,7 +895,8 @@ void WorkcellOrchestrator::_register()
       RCLCPP_INFO(this->get_logger(),
         "Successfully registered with system orchestrator");
       this->_register_timer->cancel();
-      this->_register_timer.reset();
+      // TODO(luca) reintroduce once https://github.com/ros2/rclcpp/issues/2652 is fixed and released
+      // this->_register_timer.reset();
```

Comment:
What is the implication of not resetting the timer? More CPU consumption? Can we reset it outside the service callback later on?

Should we consider switching to the events executor as suggested https://github.com/ros2/rclcpp/issues/2652#issuecomment-2419198033?

Reply:
I find the `reset` name a bit confusing in this context, since we have a pointer to a timer it can be easily misunderstood:

```
timer->reset(); // Resets the timer, but it still exists
timer.reset(); // Resets the pointer, effectively destroying the timer
```

Since we cancel the timer it will not run anymore so I believe the overhead is really just the fact that we still have memory allocated for it. I would definitely not recommend changing the whole executor for the overhead of a cancelled timer being allocated that will supposedly be fixed soon, but given how efficient the event executor seems I would consider experimenting with it and see if it helps in a future PR.

Reply:
Sounds good.